src/mpi/romio/adio/ad_lustre/ad_lustre_open.c

```diff
@@ -100,7 +100,7 @@ void ADIOI_LUSTRE_Open(ADIO_File fd, int *error_code)
             if (str_unit > UINT_MAX)
                 lum->lmm_stripe_size = UINT_MAX;
             else
-                lum->lmm_stripe_size = str_unit;
+                lum->lmm_stripe_size = (int) str_unit;
```

Comment:
Because `lum->lmm_stripe_size` is of type `__u32`,
this line of type cast can use `(unsigned int) str_unit`.

Reply:
thanks for digging into lustre internals